packages/tracking-sdk/src/main.ts

```diff
@@ -589,22 +601,81 @@ export default function main() {
 
     const mergedContext = mergeDeep(baseContext, context);
 
-    const flags = await getFlags({
+    const res = await getFlags({
       apiBaseUrl: getUrl(),
       context: mergedContext,
       timeoutMs,
       staleWhileRevalidate,
     });
 
+    let flags = res?.flags;
+
     if (!flags) {
       warn(`failed to fetch feature flags, using fall-back flags`);
-      return fallbackFlags.reduce((acc, flag) => {
+      flags = fallbackFlags.reduce((acc, flag) => {
         acc[flag.key] = flag;
         return acc;
       }, {} as Flags);
     }
 
-    return flags;
+    return proxify(
+      flags!,
+      rateLimited(FLAG_EVENTS_PER_MIN, res.url, sendCheckEvent),
```

Comment:
The flag key is not part of the `res.url`. So if you use two different flags within 1 minute, we'll only collect a check for one.

The check event should include:
- flagKey
- flagVersion: flag.version,
- evalResult
- context

and all of these should be in the key